src/shared/components/oncoprint/ResultsViewOncoprint.tsx

```diff
@@ -431,7 +449,15 @@ export default class ResultsViewOncoprint extends React.Component<IResultsViewOn
                 this.heatmapGeneInputValue = s;
                 this.heatmapGeneInputValueUpdater(); // stop updating heatmap input if user has typed
             }),
-            onSelectHeatmapProfile:(id:string)=>{this.selectedHeatmapProfile = id;},
+            onSelectHeatmapProfile:(id:string)=>onMobxPromise(
+                this.props.store.molecularProfileIdToMolecularProfile,
+                profileMap => {
+                    this.selectedHeatmapProfile = id;
+                    this.heatmapIsDynamicallyQueried = (
```

Comment:
cleaner to move this into an `autorun` that you set up in `constructor`, and remove it here and on line 164

Reply:
Since I intended this observable to list a value that should at all times indicate whether the the profile ID in `this.selectedHeatmapProfile` identifies a non-geneset profile, I think a computed property that observes both `this.selectedHeatmapProfile` and `this.props.store.molecularProfileIdToMolecularProfile` is an even better fit than an autorun reaction. I pushed a commit!